chapters/custom-license-namespaces.md

```diff
@@ -0,0 +1,194 @@
+identifier would be specified
+in a corresponding
+[Other licensing information detected](./other-licensing-information-detected.md) section.
+
+This annex defines mechanisms to use stable identifiers for custom licenses
+outside the context of a single SPDX document.
+For example, developers might wish to insert
+SPDX short-form license identifiers
+into their source code
+for licenses that are not on the SPDX License List,
+as described in
+[Annex E - Using SPDX license list short identifiers in source files](./using-SPDX-short-identifiers-in-source-files.md)
+Additionally,
+tools may want to interchange information about such licenses
+without having to use the same internal format
+to generate identifiers,
+or without having to compare license texts included in SPDX documents
+to confirm that they are in fact the same text.
+
+
+## Z.2 Brief overview
+
+This annex defines two formats and mechanisms
+for using the existing `LicenseRef-` format
+in a manner that enables stable identifier usage.
+Under this system:
+
+1. each license namespace consists of
+   either a standard order DNS name (such as `example.com`)
+   or an organization name (such as `ExampleCorp`)
+   that is under the control of the namespace maintainer's organization;
+2. the namespace maintainer creates
+   an SPDX document defining license texts
+   and corresponding identifiers
+   which include the namespave,
+   as described below;
+3. the namespace and the SPDX document
+   are registered with the SPDX project; and
+4. identifiers in the format described below
+   can then be used by anyone,
+   as long as they are defined in the SPDX document
+   available at the registered URL.
+
+
+## Z.3 License name format
+
+The following sections
+define the two formats
+for defining custom license identifiers.
+
+## Z.3.1 DNS name format
+
+A valid namespace using the DNS name format
+is a standard order DNS name,
+such as `example.com`.
+This DNS name
+must only use characters permitted by the license expression syntax
+(i.e., letters, digits, "-" and ".").
+It must not contain a period followed by a hyphen,
+as that denotes the end of the namespace.
+
+Each custom license identifier under this namespace
+consists of the following, in order:
+1. the prefix `LicenseRef-`
+2. a period
+3. the namespace
+4. a period and a hyphen
+5. any other characters permitted by the license expression syntax (i.e., letters, digits, "-" and ".")
+
+For example, for the `example.com` namespace,
+the following are identifiers:
+* `LicenseRef-.example.com.-ABC-1.0`
+* `LicenseRef-.example.com.-EULA-v3.1`
+* `LicenseRef-.example.com.-anything`
+
+## Z.3.2 Organization name format
+
+A valid namespace using the organization name format
+is the organization's name,
+such as `ExampleCorp`.
+This organization name
+must only use characters permitted by the license expression syntax
+(i.e., letters, digits, "-" and ".").
+It must not contain a double hyphen,
+as that denotes the end of the namespace.
```

Comment:
Does this lead to confusion if someone decides to implement a domain name in the organization name format, which I think would be permissible given the way this is drafted?

E.g. `LicenseRef-.example.com.-ABC-1.0` is a DNS name format, while `LicenseRef--example.com--ABC-1.0` is an organization name format -- but this may lead to confusion.

What if the organization name format were only to permit letters and digits for the namespace itself?